src/plugins/storage.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
```

Comment:
That's an interesting location. Why not just `src/` instead of `src/plugins/`? It's not like our current "plugins" folder contains actual plugins. I think we should leave it with the others in `src/`